airflow/ti_deps/deps/exec_date_after_start_date_dep.py

```diff
@@ -28,15 +28,19 @@ class ExecDateAfterStartDateDep(BaseTIDep):
 
     @provide_session
     def _get_dep_statuses(self, ti, session, dep_context):
-        if ti.task.start_date and ti.execution_date < ti.task.start_date:
+        if ti.task.start_date and ti.get_dagrun(session=session).execution_date < ti.task.start_date:
```

Comment:
Any reason why we are changing this? I think the ti.execution_date will be available at this point.

Reply:
After your commit https://github.com/apache/airflow/commit/80ae70c3d0f6109e974d50726da5a44e58d1b674 DagRun must be loaded first (via get_dagrun) or TaskInstance's execution_date would be None